cmd/main.go

```diff
@@ -179,15 +197,25 @@ func setupManager(flagVar *flags.FlagVar, cacheOptions cache.Options, scheme *ma
 		setupLog.Error(err, "unable to set up ready check")
 		os.Exit(1)
 	}
-	if flagVar.DropStoredVersion != "" {
-		go func(version string) {
-			dropStoredVersion(mgr, version)
-		}(flagVar.DropStoredVersion)
-	}
-	if err := mgr.Start(ctrl.SetupSignalHandler()); err != nil {
-		setupLog.Error(err, "problem running manager")
-		os.Exit(1)
+}
+
+func runKymaMetricsCleanup(kymaMetrics *metrics.KymaMetrics, kcpClient client.Client,
+	cleanupIntervalInMinutes time.Duration,
+) {
+	ctx, cancel := context.WithTimeout(context.Background(), metricCleanupTimeout)
+	defer cancel()
+
+	scheduler := gocron.NewScheduler(time.UTC)
+	_, scheduleErr := scheduler.Every(cleanupIntervalInMinutes).Minutes().Do(func() {
+		if err := kymaMetrics.CleanupNonExistingKymaCrsMetrics(ctx, kcpClient); err != nil {
+			setupLog.Info(fmt.Sprintf("failed to cleanup non existing kyma crs metrics, err: %s", err))
+		}
+	})
+	if scheduleErr != nil {
+		setupLog.Info(fmt.Sprintf("failed to setup cleanup routine for non existing kyma crs metrics, err: %s",
+			scheduleErr))
 	}
```

Comment:
EDIT: I think there is also a mistake in my suggestion. If we have the `defer cancel()`, the cancellation is done after running this function, so basically we still have a cancelled context on every invocation. We probably either need to omit the "global" context or provide it from the outside so that it is cancelled only once the KLM terminates

Reply:
hmm your words actually make sense. Let me think how I can fix this. Thanks for noting that!

Reply:
Maybe I am wrong, but don't you need to move the context creation in the function?
```
_, scheduleErr := scheduler.Every(cleanupIntervalInMinutes).Minutes().Do(func() {
    // Move the context creation here so it resets with each function run
    ctx, cancel := context.WithTimeout(context.Background(), metricCleanupTimeout)
    defer cancel()

    if err := kymaMetrics.CleanupNonExistingKymaCrsMetrics(ctx, kcpClient); err != nil {
        setupLog.Info(fmt.Sprintf("failed to cleanup non existing kyma crs metrics, err: %s", err))
    }
})
```